### PR TITLE
start of skeleton page

### LINKED ISSUE #1600: Skeleton - Implement first pass of site page
[Current Guidance
](https://www.figma.com/file/dQG3xZ6m4AUt7Lwrdp1gVB/Skeleton-Guidance?node-id=216%3A1663&t=m7Kpg99SjOyveF02-1) Related Ticket #3088 

## Deliverables

- [x] Add to grommet-site
- [ ] Create a first pass of the Skeleton page on the DS site. Sections to include:
     - [ ] Main example
     - [ ] When to use/when not to use
     - [ ] Anatomy

### PR DESCRIPTION
**Pull request by britt6612**
[Deploy Preview](https://deploy-preview-3077--keen-mayer-a86c8b.netlify.app/components/skeleton)
[Figma link](https://www.figma.com/file/dQG3xZ6m4AUt7Lwrdp1gVB/Skeleton-Guidance?node-id=300%3A9435&t=xB9b5v2e14DyTAU3-0)

#### What does this PR do?
Skeleton Page 
#### Where should the reviewer start?
skeleton.mdx
#### What testing has been done on this PR?
locally
In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [ ] Small, medium, and large screen sizes
- [ ] Cross-browsers (FireFox, Chrome, and Safari)
- [ ] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [ ] Keyboard interactions
- [ ] Screen reader experience
- [ ] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [ ] Console is free of warnings and errors
- [ ] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### Any background context you want to provide?
 Create a first pass of the Skeleton page on the DS site. Sections to include:
- [ ]  Main example
- [ ]  When to use/when not to use
- [ ]  Anatomy

#### What are the relevant issues?
closes #1600 
#### Should this PR be mentioned in Design System updates?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

File: aries-site/src/examples/components/skeleton/SimpleSkeleton.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import { Box, Text, Skeleton } from 'grommet';
+
+export const SkeletonExample = () => (
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/SimpleSkeleton.js:4]: You kind of hint at it in the Anatomy table, but when you do examples, keep in mind that we're trying to avoid folks using the `<Skeleton/>` component directly. Instead we're hoping they're dashboard cards or other components can just be skeleton-ized by simply using the 'skeleton' prop to 'Box' or 'Card' and let the contained 'Text', 'Heading', etc to render themselves as the gray box.

File: aries-site/src/data/structures/components.js [hunk 2]
```diff
@@ -1191,6 +1192,30 @@ export const components = [
       'spooling',
     ],
   },
+  {
+    name: 'Skeleton Loading',
```
Comment (taysea) [inline aries-site/src/data/structures/components.js:1196]:
```suggestion
    name: 'Skeleton',
```
Should this be called "Skeleton" to align with the component name?

File: aries-site/src/data/structures/components.js [hunk 3]
```diff
@@ -1191,6 +1192,30 @@ export const components = [
...
+    category: 'Visualizations',
+    description:
+      'Skeleton is used to provide a sense of the structure of the page and the ability to anticipate where certain pieces and types of content are loading. Loading patterns are used to communicate that data is loading which can improve a user experience.',
```
Comment (taysea) [inline aries-site/src/data/structures/components.js:1199]:
```suggestion
      'Skeletons tell the user that data is loading and provide a general sense of the page elements and structure. Their subtle animation conveys a sense of forward progress which can improve the user experience.',
```

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+export const DashBoardSkeletonExample = () => {
+  const [loading, setLoading] = useState(true);
+  useEffect(() => {
+    setTimeout(() => setLoading(!loading), 3000);
+  }, [loading]);
+
+  return (
+    <>
+      <Box width="large" skeleton={loading ? skeleton : undefined}>
```
Comment (taysea) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:15]: I noticed we don't have any animation on this. Is that going to be handled in a separate PR/next step?

Comment (taysea) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:15]: Should we use `Page` and `PageContent`? Does `skeleton` work on PageContent? (i'm thinking it should be PageContent is a Box.

File: aries-site/src/examples/components/card/ActivitiesNavigationalCards.js [hunk 5]
```diff
@@ -37,6 +37,7 @@ export const ActivitiesNavigationalCards = ({ heading = true }) => {
               />
             }
             level={3}
+            skeleton={skeleton}
```
Comment (taysea) [inline aries-site/src/examples/components/card/ActivitiesNavigationalCards.js:40]: I know you didn't touch it in this code, but can we sentence case "Activities and tasks" and use "and" instead of &?

File: aries-site/src/pages/components/skeleton-loading.mdx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  componentName="Skeleton"
+  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js"
+  docs="https://v2.grommet.io/skeleton?theme=hpe#props"
+>
+  <DashBoardSkeletonExample />
+</Example>
+
+### When to use
+Skeletons are used when:
+- Loading elements are known ahead of time but are not immediately available.
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton-loading.mdx:15]: When is an example of when we don't know what kind of elements we'd be rendering? It feels like generally, we know what elements will be rendered but we are waiting on the data to populate those elements

File: aries-site/src/pages/components/skeleton-loading.mdx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+- All of the content on a page is loaded in at the same time.
+- Users are loading high-traffic pages, such as a user dashboard.
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton-loading.mdx:17]: Wouldn't skeletons apply to any page? not just high-traffic ones

File: aries-site/src/pages/components/skeleton-loading.mdx [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+<Example
+  componentName="Skeleton"
+  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js"
+  docs="https://v2.grommet.io/skeleton?theme=hpe#props"
+>
+  <DashBoardSkeletonExample />
+</Example>
+
+### When to use
+Skeletons are used when:
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton-loading.mdx:14]: Not sure where this would go (probably elsewhere in the gudiance) we should mention that data should be rendered as soon as it's fetched. Using Instagram as an example, there may be multiple fetches happening (1 for stories and another for the main feed). If the stories response comes back before the main feed one, we should render the stories right away and just leave the main feed as a skeleton).

With how grommet skeleton is built, this is easy to implement by just controlling what container you flag the `skeleton` prop on. Might be a nice opportunity to highlight the ease of implementing with Grommet.

File: aries-site/src/data/structures/components.js (hunk 2 above)
Comment (britt6612) [reply aries-site/src/data/structures/components.js:1196]: good call

File: aries-site/src/examples/components/card/ActivitiesNavigationalCards.js (hunk 5 above)
Comment (britt6612) [reply aries-site/src/examples/components/card/ActivitiesNavigationalCards.js:40]: yeah I wasnt sure if this was the best to touch this file or copy over code and change it in another file

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 4 above)
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:15]: yeah I left that out while Kenny is looking more into what we want for guidance we can have it as a next step

File: aries-site/src/pages/components/skeleton-loading.mdx (hunk 7 above)
Comment (britt6612) [reply aries-site/src/pages/components/skeleton-loading.mdx:17]: @KennyAtHPE tagging for eyes

File: aries-site/src/pages/components/skeleton-loading.mdx (hunk 8 above)
Comment (britt6612) [reply aries-site/src/pages/components/skeleton-loading.mdx:14]: agree this was some feedback we got from Eric I think were going to include just not sure where but noted ✅

File: aries-site/src/pages/components/skeleton-loading.mdx (hunk 6 above)
Comment (britt6612) [reply aries-site/src/pages/components/skeleton-loading.mdx:15]: 🤔 hmm maybe we can reword @KennyAtHPE

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 4 above)
Comment (taysea) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:15]: Perfect, makes sense.

File: aries-site/src/examples/components/card/ActivitiesNavigationalCards.js (hunk 5 above)
Comment (taysea) [reply aries-site/src/examples/components/card/ActivitiesNavigationalCards.js:40]: I think we can just touch this file.

File: aries-site/src/pages/components/skeleton-loading.mdx (hunk 8 above)
Comment (taysea) [reply aries-site/src/pages/components/skeleton-loading.mdx:14]: Cool sounds good.

File: aries-site/src/pages/components/skeleton-loading.mdx [hunk 9]
```diff
@@ -0,0 +1,33 @@
+import { Notification } from 'grommet';
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton-loading.mdx:1]: We'll need to update this file name from "skeleton-loading" to "skeleton" for the routing to work

Comment (britt6612) [reply aries-site/src/pages/components/skeleton-loading.mdx]: done

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 4 above)
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js]: changed it to `page` and `pageContent` it jumps a little but let me know what you think

File: aries-site/src/pages/components/skeleton.mdx [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+  <SkeletonAnatomy />
+</Example>
+
+
+Skeletons come in three different shapes - circular, rounded, and rectangular. The shape, size, and elevation of a Skeleton are all dependent on the UI elements that are loading. 
+
+Some common components that may be skeletonized are:
+- Text, paragraph, heading
+- Avatar, image
+- Button, footer
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton.mdx:36]: I'd group "Footer" with the layout elements in the last bullet point

File: aries-site/src/pages/components/skeleton.mdx [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+- Card, table, pageheader
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton.mdx:37]: Is this list supposed to be the components that currently allow `skeleton` in Grommet? or just general? If specific, it might be nice to call this out in an implementation note and/or link to the grommet docs for skeleton

Comment (taysea) [reply aries-site/src/pages/components/skeleton.mdx:37]: Something like "Grommet has added `skeleton` property in order to make it easy to switch between a skeleton state and data-rich state for a variety of components. The components that currently support `skeleton` are:"

Comment (britt6612) [reply aries-site/src/pages/components/skeleton.mdx:37]: good point Ill make that clear

File: aries-site/src/examples/components/skeleton/SkeletonAnatomy.js [hunk 12]
```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Box, Button, Card, CardBody, Paragraph } from 'grommet';
+
+export const SkeletonAnatomy = ({ ...rest }) => (
+  <Card id="container" flex={false} width="large" {...rest}>
+    <CardBody align="start" gap="medium">
+      <Box align="center" gap="small" direction="row">
+        <Box
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/SkeletonAnatomy.js:8]: Why not use Text in a `<Box skeleton />` or at least `<Skeleton>` for these? More likely to just work if theme changes.

File: aries-site/src/examples/templates/Card/Card.js [hunk 13]
```diff
@@ -14,6 +14,18 @@ import {
 } from 'grommet';
 import { adjustPad } from './utils';
 
+//  demo purposes to make a skeleton icon for loading
+const IconSkeleton = () => (
+  <Box
```
Comment (MikeKingdom) [inline aries-site/src/examples/templates/Card/Card.js:19]: really this should instead either directly use `<Skeleton/>` or a `<Box background="whatever"/>` rather than specifying the skeleton prop (I know I'm trying to avoid specific uses of Skeleton). You should be able to avoid the conditional around where it's used if it's something the icon can be inside anyway.

File: aries-site/src/examples/templates/Card/Card.js [hunk 14]
```diff
@@ -25,13 +37,17 @@ const DesignSystemCard = ({
...
+  skeleton,
   subtitle,
   pretitle,
   title,
   ...rest
 }) => {
   const theme = useContext(ThemeContext);
 
+  // demo purposes for using skeleton
+  const skeletonAlign = skeleton ? 'none' : 'start';
```
Comment (MikeKingdom) [inline aries-site/src/examples/templates/Card/Card.js:49]: would ne nice to figure out how to avoid having to specify this and still get the right 100% behavior

File: aries-site/src/examples/templates/Card/Card.js [hunk 15]
```diff
@@ -45,14 +61,18 @@ const DesignSystemCard = ({
...
               />
             )}
-            {icon && <Box pad={{ bottom: 'small' }}>{icon}</Box>}
+            {icon && skeleton ? (
+              <IconSkeleton />
+            ) : (
+              <Box pad={{ bottom: 'small' }}>{icon}</Box>
+            )}
             {pretitle && <Text size="small">{pretitle}</Text>}
             {title && typeof title === 'string' ? (
```
Comment (MikeKingdom) [inline aries-site/src/examples/templates/Card/Card.js:77]: in the loading situation wouldn't the title be undefined? We still want the heading there in this situation so it can be a skeleton.

File: aries-site/src/data/structures/components.js [hunk 16]
```diff
@@ -1191,6 +1192,30 @@ export const components = [
...
+        background: 'background-front',
+      },
+    seoDescription:
+      'Skeletons act as placeholder previews for UI elements so that users are provided with a structure of a loading page. This helps users to anticipate where and what types of content are loading.',
+    relatedContent: ['Box', 'DataTable', 'Spinner'],
+    sections: [
+      'When to use Spinner',
+      'Spinner within a list',
+      'Spinner with announcement on screen reader',
+      'Spinner loading content',
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/data/structures/components.js]: no updated

File: aries-site/src/examples/templates/Card/Card.js (hunk 14 above)
Comment (britt6612) [reply aries-site/src/examples/templates/Card/Card.js:49]: agree I think I would have to go back and look at grommet code

File: aries-site/src/examples/templates/Card/Card.js (hunk 15 above)
Comment (britt6612) [reply aries-site/src/examples/templates/Card/Card.js:77]: Im not sure Im following,.. the `title` is still present in the skeleton render

File: aries-site/src/examples/components/skeleton/SkeletonAnatomy.js (hunk 12 above)
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/SkeletonAnatomy.js]: I wanted the `rounding` and Im not able to pass `round` to the parent `box` of `text`
so I changed to `skeleton`

File: aries-site/src/examples/templates/Card/Card.js (hunk 15 above)
Comment (MikeKingdom) [reply aries-site/src/examples/templates/Card/Card.js:75]: My point is I think it's probably working now because a title is always passed but in a real world situation you wouldn't have the title yet during loading but you still want a `<Heading>` component to provide the structure for the skeletonized view. Ie. this should still render a `<Heading>` with an empty title in a real world example

Comment (britt6612) [reply aries-site/src/examples/templates/Card/Card.js:75]: aw okay thank you sorry I wasnt sure what you meant

File: aries-site/src/examples/templates/Card/Card.js (hunk 13 above)
Comment (britt6612) [reply aries-site/src/examples/templates/Card/Card.js]: changed this let me know what you think with the `icon`

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+    setTimeout(() => setLoading(!loading), 3000);
+  }, [loading]);
+
+  return (
+    // passing skeleton to Page will skeletize all of the children that are
+    // text, button, box, paragraph, heading
+    <Page skeleton={loading ? skeleton : undefined} height="large">
+      <PageContent>
+        <Grid columns={columns[breakpoint]} gap="medium">
+          {activities.map((activity, index) => (
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:46]: I still think this needs to do something different during the 'loading' phase. It shouldn't be just iterating over the same activities. That would never happen in a real situation.

I'd do something like:
```jsx
const placeholders = [{title: 'placeholder' }, {title: 'placeholder'}, {title: 'placeholder'}];
const items = loading ? placeholders : activities;
```
then here on 46 so `items.map` rather than `activities.map`. You'll have to adjust the code to handle minimal content in the activity or put enough stuff in the placholder.

**Review by MikeKingdom — COMMENTED** [review]
Better. Can we make the grid cell height force the cards to be the same height?

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+            <Box>
```
[the remark this answers is not in the document]
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:47]: good eye no dont need I will take it out

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+                        width="24px" // size of icon
+                        height="24px"
+                        background="background"
+                        flex="grow"
+                        margin={{ bottom: 'small' }}
+                      />
+                    ) : (
+                      <Box pad={{ bottom: 'small' }}>{item?.icon}</Box>
+                    )}
+                    <Heading
```
Comment (taysea) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:72]: Screen Shot 2023-01-23 at 10 46 18 AM

it's a little strange how the padding on the right side of the card isn't there. I'd expect the heading to be a bit narrower so the padding can be there

File: aries-site/src/pages/components/skeleton.mdx [hunk 20]
```diff
@@ -0,0 +1,42 @@
...
+
+Some common components that may be skeletonized are:
+- text, paragraph, heading
+- avatar, image
+- button,
+- card, table, pageheader, footer, header
+
+Grommet has added `skeleton` property in order to make it easy to switch between a skeleton state and data-rich state for a variety of components.
+The components that currently support skeleton are:
+- box, button, heading, paragraph, text
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton.mdx:42]: Should we capitalize these since we are referring directly to the Grommet components?

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 19 above)
Comment (MikeKingdom) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:72]: That's because the Heading skeleton has a width fixed by the theme. So as the cards shrink too much they start to not have space for the heading skeleton.

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+
+  return (
+    // passing skeleton to Grid will skeletize all of the children that are
+    // text, button, box, paragraph, heading
+    <Page height="large">
+      <PageContent>
+        <Box skeleton={loading ? skeleton : undefined}>
+          <Grid columns={columns[breakpoint]} gap="medium">
+            {items.map((item, index) => {
+              console.log(item);
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:51]: should probably get rid of this now

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 19 above)
Comment (taysea) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:72]: Gotcha, wondering if we should adjust the width in theme to be something like:

```
width: { max: insert width, width: 100% }
```

I think that would allow us to set a max width and make sure the skeleton fills available space

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+              return (
+                <Card key={index}>
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:53]: I'm guessing this needs a flex="grow" to keep it from shrinking below the size needed for the heading (and maybe true for the CardHeader too.)

**Review by MikeKingdom — COMMENTED** [review]
Looking good and getting close

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 19 above)
Comment (MikeKingdom) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:72]: Heading with width 100% doesn't look good as a skeleton. It might be worth setting a better min/max in pixels though

File: aries-site/src/pages/components/skeleton.mdx (hunk 20 above)
Comment (britt6612) [reply aries-site/src/pages/components/skeleton.mdx:42]: yes I can

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 19 above)
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js]: I can play around with this in Grommet and see for a min/max.

File: aries-site/src/pages/components/skeleton.mdx (hunk 20 above)
Comment (britt6612) [reply aries-site/src/pages/components/skeleton.mdx]: done

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js (hunk 21 above)
Comment (britt6612) [reply aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js]: done

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 23]
```diff
@@ -0,0 +1,92 @@
...
+
+  return (
+    // passing skeleton to Grid will skeletize all of the children that are
+    // text, button, box, paragraph, heading
+    <Page height="large">
+      <PageContent>
+        <Box skeleton={loading ? skeleton : undefined}>
+          <Grid columns={columns[breakpoint]} gap="medium">
+            {items.map((item, index) => (
+              <Card flex="grow" key={index}>
```
Comment (MikeKingdom) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:51]: `flex="grow"` isn't sufficient here. Does it also need to be on `CardHeader` (ie. the thing containing the `Heading`)?

image

**Review by MikeKingdom — APPROVED** [review]
Looks better. Thanks!

**Comment by britt6612** [comment]
> Looks better. Thanks!

cool any thoughts on if this `min and max` should be in grommet or should we put in hpe theme?

**Comment by ericsoderberghp** [comment]
> 

I lean towards grommet

**Comment by taysea** [comment]
Screen Shot 2023-01-26 at 8 21 38 AM

Given there are only 3 cards, do we need all this extra white space at the bottom of the example? feels like we don't need the example to be so tall.

**Comment by MikeKingdom** [comment]
> Screen Shot 2023-01-26 at 8 21 38 AM
> 
> Given there are only 3 cards, do we need all this extra white space at the bottom of the example? feels like we don't need the example to be so tall.

I think the height is fixed to something that works when you shrink the width such that cards wrap to other rows. We were also trying to avoid it changing height as it flips between skeletonized and not skeletonized (since it bounces the text you're trying to read.) Thoughts on a good compromise? It looks like height=large seems a bit big but height=medium is too small for when cards wrap.

**Comment by britt6612** [comment]
> > Screen Shot 2023-01-26 at 8 21 38 AM
> > Given there are only 3 cards, do we need all this extra white space at the bottom of the example? feels like we don't need the example to be so tall.
> 
> I think the height is fixed to something that works when you shrink the width such that cards wrap to other rows. We were also trying to avoid it changing height as it flips between skeletonized and not skeletonized (since it bounces the text you're trying to read.) Thoughts on a good compromise? It looks like height=large seems a bit big but height=medium is too small for when cards wrap.

you are correct I can try `medium` but yes for this the page was bouncy when the content got loaded in which was a little distracting when trying to read the guidance.

File: aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js [hunk 24]
```diff
@@ -0,0 +1,92 @@
...
+    setTimeout(() => setLoading(!loading), 3000);
+  }, [loading]);
+
+  const placeholders = [{}, {}, {}];
+  const items = loading ? placeholders : activities;
+
+  return (
+    // passing skeleton to Grid will skeletize all of the children that are
+    // text, button, box, paragraph, heading
+    <Page height="large">
```
Comment (taysea) [inline aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js:46]:
```suggestion
    <Page>
```
Instead of setting the height here, we can set it on the "Example" container and do a custom value (since it won't be exposed)

File: aries-site/src/pages/components/skeleton.mdx [hunk 25]
```diff
@@ -0,0 +1,37 @@
+import { Notification } from 'grommet';
+import { Example } from '../../layouts';
+import { DashBoardSkeletonExample, SkeletonAnatomy } from '../../examples';
+
+<Example
+  componentName="Skeleton"
+  code="https://raw.githubusercontent.com/grommet/hpe-design-system/master/aries-site/src/examples/components/skeleton/DashBoardSkeletonExample.js"
+  docs="https://v2.grommet.io/skeleton?theme=hpe#props"
+  width="large"
```
Comment (taysea) [inline aries-site/src/pages/components/skeleton.mdx:9]:
```suggestion
template
height="500px"
```
^^ this should allow us to set a reasonable height that won't jump. See if this works.

**Comment by taysea** [comment]
> > > Given there are only 3 cards, do we need all this extra white space at the bottom of the example? feels like we don't need the example to be so tall.
> > 
> > 
> > I think the height is fixed to something that works when you shrink the width such that cards wrap to other rows. We were also trying to avoid it changing height as it flips between skeletonized and not skeletonized (since it bounces the text you're trying to read.) Thoughts on a good compromise? It looks like height=large seems a bit big but height=medium is too small for when cards wrap.
> 
> you are correct I can try `medium` but yes for this the page was bouncy when the content got loaded in which was a little distracting when trying to read the guidance.

I've dropped a few comments about how we might handle this in the "Example" height and do a custom value. This cleans things up so we don't have a fixed height in the example code itself (since we wouldn't want people copy and pasting that into their app anyway).

**Comment by britt6612** [comment]
> > > > Given there are only 3 cards, do we need all this extra white space at the bottom of the example? feels like we don't need the example to be so tall.
> > > 
> > > 
> > > I think the height is fixed to something that works when you shrink the width such that cards wrap to other rows. We were also trying to avoid it changing height as it flips between skeletonized and not skeletonized (since it bounces the text you're trying to read.) Thoughts on a good compromise? It looks like height=large seems a bit big but height=medium is too small for when cards wrap.
> > 
> > 
> > you are correct I can try `medium` but yes for this the page was bouncy when the content got loaded in which was a little distracting when trying to read the guidance.
> 
> I've dropped a few comments about how we might handle this in the "Example" height and do a custom value. This cleans things up so we don't have a fixed height in the example code itself (since we wouldn't want people copy and pasting that into their app anyway).

sounds good Ill check out the changes you suggested

**Comment by MikeKingdom** [comment]
> > I've dropped a few comments about how we might handle this in the "Example" height and do a custom value. This cleans things up so we don't have a fixed height in the example code itself (since we wouldn't want people copy and pasting that into their app anyway).
> 
> sounds good Ill check out the changes you suggested

Thanks Taylor. I learned a long time ago assume people will cut and paste everything without learning much about it so good to think of that ahead of time :)

**Review by taysea — APPROVED** [review]
LGTM, thanks for all of iterations!
